src/eppo_value.ts

```diff
@@ -32,38 +33,39 @@ export class EppoValue {
   }
 
   static generateEppoValue(
-    expectedValueType?: ValueType,
-    value?: string,
-    typedValue?: boolean | number | string | object,
+    value: boolean | number | string | object,
+    valueType: VariationType,
   ): EppoValue {
-    if (value != null && typedValue != null) {
-      switch (expectedValueType) {
-        case ValueType.BoolType:
-          return EppoValue.Bool(typedValue as boolean);
-        case ValueType.NumericType:
-          return EppoValue.Numeric(typedValue as number);
-        case ValueType.StringType:
-          return EppoValue.String(typedValue as string);
-        case ValueType.JSONType:
-          return EppoValue.JSON(value, typedValue as object);
-        default:
-          return EppoValue.String(value as string);
-      }
+    if (value == null) {
+      return EppoValue.Null();
+    }
+    switch (valueType) {
+      case VariationType.BOOLEAN:
+        return EppoValue.Bool(value as boolean);
+      case VariationType.NUMERIC:
+        return EppoValue.Numeric(value as number);
+      case VariationType.INTEGER:
+        return EppoValue.Numeric(value as number);
```

Comment:
minor, but using `case` magic (i.e., they don't break by default) this can be simplified to:
```ts
case VariationType.NUMERIC:
case VariationType.INTEGER:
  return EppoValue.Numeric(value as number);
```

Reply:
Actually, I think we need to nix the concept of EppoValueType, and go back to a singular `ValueType` because its important for EppoValue to know if its an INTEGER or NUMERIC for type checking

Reply:
May also want to do `EppoValue.Numeric(Math.round(value as number))` to ensure we just store an integer, although really that should be in a to-be-created `EppoValue.Integer()`

Reply:
And per our slack conversation, we also could just ax `EppoValue` entirely 🪓 
This would be my vote 😅

Reply:
Going to vote against the `case` magic: it's saving one line but harms readability in my opinion, and might confuse readers that there is a bug.